gpytorch/lazy/cached_cg_lazy_tensor.py

```diff
@@ -134,19 +134,21 @@ def requires_grad(self):
     def requires_grad(self, val):
         self.base_lazy_tensor.requires_grad = val
 
-    def _cholesky(self):
+    def _cholesky(self, upper=False):
+        from .triangular_lazy_tensor import TriangularLazyTensor
+
         res = self.__class__(
-            self.base_lazy_tensor._cholesky(),
+            self.base_lazy_tensor._cholesky(upper=upper),
             eager_rhss=self.eager_rhss,
             solves=self.solves,
             probe_vectors=self.probe_vectors,
             probe_vector_norms=self.probe_vector_norms,
             probe_vector_solves=self.probe_vector_solves,
             probe_vector_tmats=self.probe_vector_tmats,
         )
-        return res
+        return TriangularLazyTensor(res, upper=upper)
```

Comment:
And here

Reply:
I think here we want `TriangularLazyTensor` wrapping the CachedCGLazyTensor, but maybe could remove the innr TriangularLazyTensor.